CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/utils/MediaStoreUtils.kt

```diff
@@ -0,0 +1,95 @@
+import android.os.Build
+import android.provider.MediaStore
+import androidx.core.net.toUri
+import java.io.File
+
+
+/**
+ * A utility class for accessing this app's photo storage.
+ *
+ * Since this app doesn't request any external storage permissions, it will only be able to access
+ * photos taken with this app. If the app is uninstalled, the photos taken with this app will stay
+ * on the device, but reinstalling the app will not give it access to photos taken with the app's
+ * previous instance. You can request further permissions to change this app's access. See this
+ * guide for more: https://developer.android.com/training/data-storage.
+ */
+class MediaStoreUtils(private val context: Context) {
+
+    // Suppress DATA index deprecation warning since we need the file location for the Glide library
+    @Suppress("DEPRECATION")
+    private val imageDataColumnIndex = MediaStore.Images.Media.DATA
+    private val imageIdColumnIndex = MediaStore.Images.Media._ID
+    val mediaStoreCollection: Uri? = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+        MediaStore.Images.Media.getContentUri(MediaStore.VOLUME_EXTERNAL)
+    } else {
+        context.getExternalFilesDir(null)?.toUri()
+    }
+
+    private fun getMediaStoreImageCursor(mediaStoreCollection: Uri): Cursor? {
+        val projection = arrayOf(imageDataColumnIndex, imageIdColumnIndex)
+        val sortOrder = "DATE_ADDED DESC"
+        return context.contentResolver.query(
+            mediaStoreCollection,  projection, null, null, sortOrder)
+    }
+
+    fun getLatestImageFilename(): String? {
+        if (mediaStoreCollection == null) return null
+
+        val cursor = getMediaStoreImageCursor(mediaStoreCollection)
+        if (cursor?.moveToFirst() != true) return null
+
+        val filename = cursor.getString(cursor.getColumnIndexOrThrow(imageDataColumnIndex))
+        cursor.close()
+
+        return filename
+    }
+
+    fun getImages(): MutableList<MediaStoreFile> {
+        val files = mutableListOf<MediaStoreFile>()
+        if (mediaStoreCollection == null) return files
+
+        val cursor = getMediaStoreImageCursor(mediaStoreCollection)
+        val imageDataColumn = cursor?.getColumnIndexOrThrow(imageDataColumnIndex)
+        val imageIdColumn = cursor?.getColumnIndexOrThrow(imageIdColumnIndex)
+
+        if (cursor != null && imageDataColumn != null && imageIdColumn != null) {
+            while (cursor.moveToNext()) {
+                val id = cursor.getLong(imageIdColumn)
+                val contentUri: Uri = ContentUris.withAppendedId(
+                    MediaStore.Images.Media.EXTERNAL_CONTENT_URI,
+                    id
+                )
+                val contentFile = File(cursor.getString(imageDataColumn))
+                files.add(MediaStoreFile(contentUri, contentFile, id))
+            }
+        }
+
+        cursor?.close()
+
+        return files
+    }
+}
+
+class MediaStoreFile(val uri: Uri, val file: File, val id: Long)
```

Comment:
should this be a data class or was it intentional that equals / hashcode / toString wasn't needed?

Reply:
Just an oversight. Updated.